src/allmydata/crypto/ed25519.py

```diff
@@ -0,0 +1,127 @@
+'''
+Ed25519 keys and helpers.
+
+Key Formatting
+--------------
+
+- in base32, keys are 52 chars long (both signing and verifying keys)
+- in base62, keys is 43 chars long
+- in base64, keys is 43 chars long
+
+We can't use base64 because we want to reserve punctuation and preserve
+cut-and-pasteability. The base62 encoding is shorter than the base32 form,
+but the minor usability improvement is not worth the documentation and
+specification confusion of using a non-standard encoding. So we stick with
+base32.
+'''
+
+import six
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey, Ed25519PublicKey
+from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, NoEncryption, \
+    PublicFormat
+
+from allmydata.crypto import remove_prefix, BadSignature
+from allmydata.util.base32 import a2b, b2a
+
+_PRIV_PREFIX = 'priv-v0-'
+_PUB_PREFIX = 'pub-v0-'
+
+
+class SigningKey:
+
```

Comment:
(See discussion about "wrapper classes" ..) but if this stays, docstrings please.